backend/windmill-api/src/job_helpers.rs

```
@@ -219,11 +238,157 @@ async fn list_stored_datasets(
         }
     }
 
+    let file_count = stored_datasets.len();
+    #[cfg(not(feature = "enterprise"))]
+    if stored_datasets.len() > 10 {
+        stored_datasets = vec![];
```

should return an error there